### PR TITLE
Adds `displayFontFamily` and `bodyFontFamily` to TextTheme apply.

### LINKED ISSUE #148510: Add `bodyFontFamily` and `displayFontFamily` to `TextTheme.apply()`
### Use case

When calling `ThemeData.apply()` we are able to define two different colors, separating base text from emphasized large text.

From https://stackoverflow.com/a/49287410/1540350:

> `bodyColor` will be applied to `headline`, `title`, `subhead`, `button`, `body1`, and `body2`. `displayColor` will be applied to `display1` through `display4`, and `caption`. If you specify both `bodyColor` and `displayColor` and use the same color value, that will effectively change text colors on all text styles.
> 
> Example:
> ```dart
> final newTextTheme = Theme.of(context).textTheme.apply(
>   bodyColor: Colors.pink,
>   displayColor: Colors.pink,
> );

I like this separation and would enjoy to have the same consistent experience for the font family, however there is only `apply`'s `fontFamily`, so in order to make this separation with fonts, I have to construct some crazy tree:

```dart
  static ThemeData materialLightTheme = ThemeData.light().copyWith(
      colorScheme: ColorScheme.fromSeed(seedColor: envPrimaryColor),
      textTheme: ThemeData.light()
          .textTheme
          .apply(
            fontFamily: envBodyFont,
          )
          .copyWith(
              displaySmall: ThemeData.light().textTheme.displaySmall!.copyWith(
                    fontFamily: envDisplayFont,
                  ),
              displayMedium:
                  ThemeData.light().textTheme.displayMedium!.copyWith(
                        fontFamily: envDisplayFont,
                      ),
              displayLarge: ThemeData.light().textTheme.displayLarge!.copyWith(
                    fontFamily: envDisplayFont,
                  )),
      cupertinoOverrideTheme: CupertinoThemeData(
        primaryColor: envPprimaryColor,
      ));
```

Ugh, I'm not a fan of that.

### Proposal

Something like this would satisfy me:

```dart
final newTextTheme = Theme.of(context).textTheme.apply(
  bodyFontFamily: envBodyFont,
  displayFontFamily: envDisplayFont,
);
```

Only if the specific value is not set, it should fallback to the more generic `fontFamily` to avoid breaking changes.

Thank you!

### PR DESCRIPTION
**Pull request by chinmoy12c**
This PR adds `displayFontFamily` and `bodyFontFamily` to `TextTheme.apply()` method.

Fixes: #148510 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/text_theme.dart [hunk 1]
```diff
@@ -355,12 +355,20 @@ class TextTheme with Diagnosticable {
   /// [displaySmall], [headlineLarge], [headlineMedium], and [bodySmall]. The
   /// `bodyColor` is applied to the remaining text styles.
   ///
+  /// The `displayFontFamily` is applied to [displayLarge], [displayMedium],
+  /// [displaySmall], [headlineLarge], [headlineMedium], and [bodySmall]. The
```
Comment (Piinks) [inline packages/flutter/lib/src/material/text_theme.dart:359]: Why would display apply to bodySmall?

File: packages/flutter/lib/src/material/text_theme.dart [hunk 2]
```diff
@@ -355,12 +355,20 @@ class TextTheme with Diagnosticable {
...
+  /// `bodyFontFamily` is applied to the remaining text styles.
```
Comment (Piinks) [inline packages/flutter/lib/src/material/text_theme.dart:360]: What are the remaining style? And how do these relate to fontFamily and fontFamilyFallback?

File: packages/flutter/lib/src/material/text_theme.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/material/text_theme.dart:359]: And how does it relate to the headline ones as well?

**Review by Piinks — COMMENTED** [review]
I am not sure this is a change we should make. The new display and body properties here don't seem to relate to all the text styles they are applied to.
If we add individual font families for display and body, does that mean we might also have to add them for headline, title, and label? Or what is folks want to specify a font family for each variant of small, medium and large?

I don't know that we want to enumerate every option on this method, since at that point folks would probably be better configuring their TextTheme another way.

**Comment by Piinks** [comment]
@chinmoy12c greetings from triage, would you like to continue working on this change?

File: packages/flutter/lib/src/material/text_theme.dart (hunk 1 above)
Comment (chinmoy12c) [reply packages/flutter/lib/src/material/text_theme.dart:359]: Hey this was done basis what `displayColor` follows. Based on the reply I received here: https://discord.com/channels/608014603317936148/608021234516754444/1242440562452332565 I assume this is not relevant now?

File: packages/flutter/lib/src/material/text_theme.dart (hunk 2 above)
Comment (chinmoy12c) [reply packages/flutter/lib/src/material/text_theme.dart:360]: The other styles include
`headlineSmall`
`titleLarge`
`titleMedium`
`titleSmall`
`bodyLarge`
`bodyMedium`
`bodySmall`
`labelLarge`
`labelMedium`
`labelSmall`

This is the config that `displayColor` follows.

**Comment by chinmoy12c** [comment]
> @chinmoy12c greetings from triage, would you like to continue working on this change?

Yes, sorry for the delay. Just a bit crunched on time rn 😅

**Comment by Piinks** [comment]
Hey @chinmoy12c, since we have not heard anything here for a while, I'm going to close this PR for now to remove it from our PR review queue. Please don't hesitate to re-open or submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**